gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -274,7 +274,14 @@
    * Configuration properties for the data publisher.
    */
   public static final String DATA_PUBLISHER_PREFIX = "data.publisher";
+
+  /**
+   * @deprecated Use {@link #TASK_DATA_PUBLISHER_TYPE} and {@link #JOB_DATA_PUBLISHER_TYPE}.
+   */
+  @Deprecated
   public static final String DATA_PUBLISHER_TYPE = DATA_PUBLISHER_PREFIX + ".type";
+  public static final String JOB_DATA_PUBLISHER_TYPE = "job." + DATA_PUBLISHER_PREFIX + ".type";
+  public static final String TASK_DATA_PUBLISHER_TYPE = "task." + DATA_PUBLISHER_PREFIX + ".type";
```

Comment:
Can you switch the order of `task.` and `DATA_PUBLISHER_PREFIX`, otherwise its not really a prefix. Same for the the `job` param above.